lib/smart_proxy_salt/api_request.rb

```diff
@@ -1,44 +1,47 @@
+# frozen_string_literal: true
+
 require 'json'
 require 'net/http'
 require 'net/https'
 require 'uri'
 
-module Proxy::Salt
-  class ApiError < RuntimeError; end
-
-  class ApiRequest
-    attr_reader :url, :username, :password, :auth
-
-    def initialize
-      @url = Proxy::Salt::Plugin.settings.api_url
-      @auth = Proxy::Salt::Plugin.settings.api_auth
-      @username = Proxy::Salt::Plugin.settings.api_username
-      @password = Proxy::Salt::Plugin.settings.api_password
-
-      begin
-        URI.parse(url)
-      rescue URI::InvalidURIError => e
-        raise ConfigurationError.new("Invalid Salt api_url setting: #{e}")
+module Proxy
+  module Salt
+    class ApiError < RuntimeError; end
+
+    # SaltStack's Rest API
+    class ApiRequest
+      attr_reader :url, :username, :password, :auth
+
+      def initialize
+        @url = Proxy::Salt::Plugin.settings.api_url
+        @auth = Proxy::Salt::Plugin.settings.api_auth
+        @username = Proxy::Salt::Plugin.settings.api_username
+        @password = Proxy::Salt::Plugin.settings.api_password
+
+        begin
+          URI.parse(url)
+        rescue URI::InvalidURIError => e
+          raise ConfigurationError.new("Invalid Salt api_url setting: #{e}")
+        end
       end
-    end
 
-    def post(path, options = {})
-      uri              = URI.parse(url)
-      http             = Net::HTTP.new(uri.host, uri.port)
-      http.use_ssl     = uri.scheme == 'https'
-      http.verify_mode = OpenSSL::SSL::VERIFY_NONE
-      path = [uri.path, path].join('/') unless uri.path.empty?
+      def post(path, options = {})
+        uri              = URI.parse(url)
+        http             = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl     = uri.scheme == 'https'
+        http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+        path = [uri.path, path].join('/') unless uri.path.empty?
```

Comment:
IMHO this is the reason you need to mock `//run` in the `test/unit/salt_test_rest_test.rb`, because the `path` that is supplied by the methods in `lib/smart_proxy_salt/rest.rb` already includes one `/` and the `join()` adds another one.

Reply:
right. should be part of another PR then.

Reply:
Fixed. Thanks.